SBB - guild is null in some cases

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:115)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:91)
        at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:53)
        at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:58)
Caused by: java.lang.NullPointerException: Cannot invoke "net.dv8tion.jda.api.entities.Guild.updateCommands()" because the return value of "de.rettichlp.shantybot.common.configuration.DiscordBotProperties.getGuild()" is null
        at de.rettichlp.shantybot.ShantyBot.startDiscordBot(ShantyBot.java:77)
        at de.rettichlp.shantybot.ShantyBot.main(ShantyBot.java:50)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 4 more
Exception in thread "Timer-1" java.lang.NullPointerException: Cannot invoke "net.dv8tion.jda.api.entities.Guild.getMemberCount()" because the return value of "de.rettichlp.shantybot.common.configuration.DiscordBotProperties.getGuild()" is null
        at de.rettichlp.shantybot.common.services.ActivitySyncService$1.run(ActivitySyncService.java:33)
        at java.base/java.util.TimerThread.mainLoop(Timer.java:566)
        at java.base/java.util.TimerThread.run(Timer.java:516)
```
```
Exception in thread "Timer-0" java.lang.NullPointerException: Cannot invoke "net.dv8tion.jda.api.entities.Guild.loadMembers()" because "guild" is null
        at de.rettichlp.shantybot.common.services.RoleSyncService$1.run(RoleSyncService.java:29)
        at java.base/java.util.TimerThread.mainLoop(Timer.java:566)
        at java.base/java.util.TimerThread.run(Timer.java:516)
```
```
java.lang.NullPointerException: Cannot invoke "net.dv8tion.jda.api.entities.Guild.getSelfMember()" because "guild" is null
        at de.rettichlp.shantybot.buttons.ResumeButton.onButtonClick(ResumeButton.java:32) ~[!/:1.1.0]
        at de.rettichlp.shantybot.buttons.ButtonBase.onButtonInteraction(ButtonBase.java:20) ~[!/:1.1.0]
        at net.dv8tion.jda.api.hooks.ListenerAdapter.onEvent(ListenerAdapter.java:456) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.api.hooks.InterfacedEventManager.handle(InterfacedEventManager.java:98) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.hooks.EventManagerProxy.handleInternally(EventManagerProxy.java:88) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.hooks.EventManagerProxy.handle(EventManagerProxy.java:70) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.JDAImpl.handleEvent(JDAImpl.java:169) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.handle.InteractionCreateHandler.handleAction(InteractionCreateHandler.java:134) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.handle.InteractionCreateHandler.handleInternally(InteractionCreateHandler.java:86) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.handle.SocketHandler.handle(SocketHandler.java:39) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.requests.WebSocketClient.onDispatch(WebSocketClient.java:1009) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.requests.WebSocketClient.onEvent(WebSocketClient.java:892) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.requests.WebSocketClient.handleEvent(WebSocketClient.java:870) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.requests.WebSocketClient.onBinaryMessage(WebSocketClient.java:1048) ~[JDA-5.0.2.jar!/:5.0.2]
        at com.neovisionaries.ws.client.ListenerManager.callOnBinaryMessage(ListenerManager.java:385) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.callOnBinaryMessage(ReadingThread.java:276) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.handleBinaryFrame(ReadingThread.java:996) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.handleFrame(ReadingThread.java:755) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.main(ReadingThread.java:108) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.runMain(ReadingThread.java:64) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.WebSocketThread.run(WebSocketThread.java:45) ~[nv-websocket-client-2.14.jar!/:na]
```
```
java.lang.NullPointerException: Cannot invoke "net.dv8tion.jda.api.entities.Guild.getSelfMember()" because "guild" is null
        at de.rettichlp.shantybot.buttons.StopButton.onButtonClick(StopButton.java:31) ~[!/:1.1.0]
        at de.rettichlp.shantybot.buttons.ButtonBase.onButtonInteraction(ButtonBase.java:20) ~[!/:1.1.0]
        at net.dv8tion.jda.api.hooks.ListenerAdapter.onEvent(ListenerAdapter.java:456) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.api.hooks.InterfacedEventManager.handle(InterfacedEventManager.java:98) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.hooks.EventManagerProxy.handleInternally(EventManagerProxy.java:88) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.hooks.EventManagerProxy.handle(EventManagerProxy.java:70) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.JDAImpl.handleEvent(JDAImpl.java:169) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.handle.InteractionCreateHandler.handleAction(InteractionCreateHandler.java:134) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.handle.InteractionCreateHandler.handleInternally(InteractionCreateHandler.java:86) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.handle.SocketHandler.handle(SocketHandler.java:39) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.requests.WebSocketClient.onDispatch(WebSocketClient.java:1009) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.requests.WebSocketClient.onEvent(WebSocketClient.java:892) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.requests.WebSocketClient.handleEvent(WebSocketClient.java:870) ~[JDA-5.0.2.jar!/:5.0.2]
        at net.dv8tion.jda.internal.requests.WebSocketClient.onBinaryMessage(WebSocketClient.java:1048) ~[JDA-5.0.2.jar!/:5.0.2]
        at com.neovisionaries.ws.client.ListenerManager.callOnBinaryMessage(ListenerManager.java:385) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.callOnBinaryMessage(ReadingThread.java:276) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.handleBinaryFrame(ReadingThread.java:996) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.handleFrame(ReadingThread.java:755) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.main(ReadingThread.java:108) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.ReadingThread.runMain(ReadingThread.java:64) ~[nv-websocket-client-2.14.jar!/:na]
        at com.neovisionaries.ws.client.WebSocketThread.run(WebSocketThread.java:45) ~[nv-websocket-client-2.14.jar!/:na]
```

--- a/src/main/java/de/rettichlp/shantybot/ShantyBot.java
+++ b/src/main/java/de/rettichlp/shantybot/ShantyBot.java
@@ -73,14 +73,14 @@ public class ShantyBot implements WebMvcConfigurer {
                 )
                 .build().awaitReady();
 
-        discordBotProperties.getGuild().updateCommands().addCommands(
+        discordBot.getGuilds().forEach(guild -> guild.updateCommands().addCommands(
                 slash("play", "Lässt den Bot Deinen Channel betreten und die angegebene Musik spielen")
                         .addOption(STRING, "link", "Link oder Name des Songs", true),
                 slash("version", "Zeigt die aktuelle Version des ShantyBots"),
                 slash("löschen", "Löscht die angegebene Menge an Nachrichten (optional eines bestimmten Nutzers)")
                         .addOption(INTEGER, "anzahl", "Anzahl der Nachrichten, die gelöscht werden sollen", true)
                         .setDefaultPermissions(enabledFor(MESSAGE_MANAGE))
-        ).queue();
+        ).queue());
 
         audioPlayerManager = new AudioPlayerManager();
     }

--- a/src/main/java/de/rettichlp/shantybot/buttons/PauseButton.java
+++ b/src/main/java/de/rettichlp/shantybot/buttons/PauseButton.java
@@ -9,10 +9,10 @@ import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;
 import java.util.Objects;
 
 import static de.rettichlp.shantybot.ShantyBot.audioPlayerManager;
-import static de.rettichlp.shantybot.ShantyBot.discordBotProperties;
 import static de.rettichlp.shantybot.common.services.UtilService.sendSelfDeletingMessage;
 import static java.util.Objects.isNull;
 import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
 
 public class PauseButton extends ButtonBase {
 
@@ -28,7 +28,7 @@ public class PauseButton extends ButtonBase {
             return;
         }
 
-        Guild guild = discordBotProperties.getGuild();
+        Guild guild = ofNullable(event.getGuild()).orElseThrow(() -> new NullPointerException("Guild is null"));
         GuildVoiceState selfVoiceState = guild.getSelfMember().getVoiceState();
         if (isNull(selfVoiceState) || !selfVoiceState.inAudioChannel()) {
             sendSelfDeletingMessage(event, "Es wird gerade keine Musik gespielt!");

--- a/src/main/java/de/rettichlp/shantybot/buttons/ResumeButton.java
+++ b/src/main/java/de/rettichlp/shantybot/buttons/ResumeButton.java
@@ -9,10 +9,10 @@ import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;
 import java.util.Objects;
 
 import static de.rettichlp.shantybot.ShantyBot.audioPlayerManager;
-import static de.rettichlp.shantybot.ShantyBot.discordBotProperties;
 import static de.rettichlp.shantybot.common.services.UtilService.sendSelfDeletingMessage;
 import static java.util.Objects.isNull;
 import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
 
 public class ResumeButton extends ButtonBase {
 
@@ -28,7 +28,7 @@ public class ResumeButton extends ButtonBase {
             return;
         }
 
-        Guild guild = discordBotProperties.getGuild();
+        Guild guild = ofNullable(event.getGuild()).orElseThrow(() -> new NullPointerException("Guild is null"));
         GuildVoiceState selfVoiceState = guild.getSelfMember().getVoiceState();
         if (isNull(selfVoiceState) || !selfVoiceState.inAudioChannel()) {
             sendSelfDeletingMessage(event, "Es wird gerade keine Musik gespielt!");

--- a/src/main/java/de/rettichlp/shantybot/buttons/SkipButton.java
+++ b/src/main/java/de/rettichlp/shantybot/buttons/SkipButton.java
@@ -8,10 +8,10 @@ import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;
 import java.util.Objects;
 
 import static de.rettichlp.shantybot.ShantyBot.audioPlayerManager;
-import static de.rettichlp.shantybot.ShantyBot.discordBotProperties;
 import static de.rettichlp.shantybot.common.services.UtilService.sendSelfDeletingMessage;
 import static java.util.Objects.isNull;
 import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
 
 public class SkipButton extends ButtonBase {
 
@@ -27,7 +27,7 @@ public class SkipButton extends ButtonBase {
             return;
         }
 
-        Guild guild = discordBotProperties.getGuild();
+        Guild guild = ofNullable(event.getGuild()).orElseThrow(() -> new NullPointerException("Guild is null"));
         GuildVoiceState selfVoiceState = guild.getSelfMember().getVoiceState();
         if (isNull(selfVoiceState) || !selfVoiceState.inAudioChannel()) {
             sendSelfDeletingMessage(event, "Es wird gerade keine Musik gespielt!");

--- a/src/main/java/de/rettichlp/shantybot/buttons/StopButton.java
+++ b/src/main/java/de/rettichlp/shantybot/buttons/StopButton.java
@@ -8,10 +8,10 @@ import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;
 import java.util.Objects;
 
 import static de.rettichlp.shantybot.ShantyBot.audioPlayerManager;
-import static de.rettichlp.shantybot.ShantyBot.discordBotProperties;
 import static de.rettichlp.shantybot.common.services.UtilService.sendSelfDeletingMessage;
 import static java.util.Objects.isNull;
 import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
 
 public class StopButton extends ButtonBase {
 
@@ -27,7 +27,7 @@ public class StopButton extends ButtonBase {
             return;
         }
 
-        Guild guild = discordBotProperties.getGuild();
+        Guild guild = ofNullable(event.getGuild()).orElseThrow(() -> new NullPointerException("Guild is null"));
         GuildVoiceState selfVoiceState = guild.getSelfMember().getVoiceState();
         if (isNull(selfVoiceState) || !selfVoiceState.inAudioChannel()) {
             sendSelfDeletingMessage(event, "Es wird gerade keine Musik gespielt!");

--- a/src/main/java/de/rettichlp/shantybot/commands/MusicPlayCommand.java
+++ b/src/main/java/de/rettichlp/shantybot/commands/MusicPlayCommand.java
@@ -8,11 +8,11 @@ import net.dv8tion.jda.api.interactions.commands.OptionMapping;
 import net.dv8tion.jda.api.managers.AudioManager;
 
 import static de.rettichlp.shantybot.ShantyBot.audioPlayerManager;
-import static de.rettichlp.shantybot.ShantyBot.discordBotProperties;
 import static de.rettichlp.shantybot.common.services.UtilService.sendSelfDeletingMessage;
 import static java.lang.String.join;
 import static java.util.Objects.isNull;
 import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
 
 public class MusicPlayCommand extends CommandBase {
 
@@ -34,7 +34,7 @@ public class MusicPlayCommand extends CommandBase {
             return;
         }
 
-        Guild guild = discordBotProperties.getGuild();
+        Guild guild = ofNullable(event.getGuild()).orElseThrow(() -> new NullPointerException("Guild is null"));
         GuildVoiceState selfVoiceState = guild.getSelfMember().getVoiceState();
         if (isNull(selfVoiceState) || !selfVoiceState.inAudioChannel()) {
             AudioManager audioManager = guild.getAudioManager();

--- a/src/main/java/de/rettichlp/shantybot/common/configuration/DiscordBotProperties.java
+++ b/src/main/java/de/rettichlp/shantybot/common/configuration/DiscordBotProperties.java
@@ -4,10 +4,12 @@ import lombok.Getter;
 import net.dv8tion.jda.api.entities.Guild;
 import net.dv8tion.jda.api.entities.Role;
 import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
+import org.jetbrains.annotations.Nullable;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
 import static de.rettichlp.shantybot.ShantyBot.discordBot;
+import static java.util.Optional.ofNullable;
 
 @Getter
 @Component
@@ -28,15 +30,22 @@ public class DiscordBotProperties {
     @Value("${discord.guild.roles.member-role}")
     private String memberRole;
 
+    @Nullable
     public Guild getGuild() {
         return discordBot.getGuildById(this.guildId);
     }
 
+    @Nullable
     public TextChannel getCommunityTextChannel() {
-        return getGuild().getTextChannelById(this.communityTextChannel);
+        return ofNullable(getGuild())
+                .map(guild -> guild.getTextChannelById(this.communityTextChannel))
+                .orElse(null);
     }
 
+    @Nullable
     public Role getMemberRole() {
-        return getGuild().getRoleById(this.memberRole);
+        return ofNullable(getGuild())
+                .map(guild -> guild.getRoleById(this.memberRole))
+                .orElse(null);
     }
 }

--- a/src/main/java/de/rettichlp/shantybot/common/services/ActivitySyncService.java
+++ b/src/main/java/de/rettichlp/shantybot/common/services/ActivitySyncService.java
@@ -23,7 +23,7 @@ public class ActivitySyncService {
 
     private int index = 0;
 
-    @Scheduled(cron = "0 0/5 * * * ?")
+    @Scheduled(cron = "0 */2 * * * *") // every day, every 2 minutes
     public void updateActivity() {
         String newActivity = this.activities[this.index].replace("%count%", getMemberCount());
         discordBot.getPresence().setActivity(playing(newActivity));

--- a/src/main/java/de/rettichlp/shantybot/common/services/ActivitySyncService.java
+++ b/src/main/java/de/rettichlp/shantybot/common/services/ActivitySyncService.java
@@ -1,19 +1,17 @@
 package de.rettichlp.shantybot.common.services;
 
 import lombok.extern.log4j.Log4j2;
-import net.dv8tion.jda.api.entities.Activity;
+import net.dv8tion.jda.api.entities.Guild;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
-
-import java.util.Timer;
-import java.util.TimerTask;
 
 import static de.rettichlp.shantybot.ShantyBot.discordBot;
 import static de.rettichlp.shantybot.ShantyBot.discordBotProperties;
-import static java.lang.String.valueOf;
-import static java.lang.System.currentTimeMillis;
-import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.Optional.ofNullable;
 import static net.dv8tion.jda.api.entities.Activity.playing;
 
+@EnableScheduling
 @Log4j2
 @Component
 public class ActivitySyncService {
@@ -23,18 +21,18 @@ public class ActivitySyncService {
             "mit %count% Spielern",
     };
 
-    public ActivitySyncService() {
-        long fiveMinutesInMillis = MINUTES.toMillis(5);
+    private int index = 0;
 
-        new Timer().scheduleAtFixedRate(new TimerTask() {
-            @Override
-            public void run() {
-                String activityString = ActivitySyncService.this.activities[(int) (currentTimeMillis() / fiveMinutesInMillis) % ActivitySyncService.this.activities.length];
+    @Scheduled(cron = "0 0/5 * * * ?")
+    public void updateActivity() {
+        String newActivity = this.activities[this.index].replace("%count%", getMemberCount());
+        discordBot.getPresence().setActivity(playing(newActivity));
+        this.index = (this.index + 1) % this.activities.length;
+    }
 
-                Activity activity = playing(activityString.replace("%count%", valueOf(discordBotProperties.getGuild().getMemberCount())));
-                discordBot.getPresence().setActivity(activity);
-            }
-        }, fiveMinutesInMillis - currentTimeMillis() % fiveMinutesInMillis, fiveMinutesInMillis);
-        log.info("Activity synchronising: scheduled");
+    private String getMemberCount() {
+        return ofNullable(discordBotProperties.getGuild())
+                .map(Guild::getMemberCount)
+                .map(Object::toString).orElse("n/a");
     }
 }

--- a/src/main/java/de/rettichlp/shantybot/common/services/RoleSyncService.java
+++ b/src/main/java/de/rettichlp/shantybot/common/services/RoleSyncService.java
@@ -1,42 +1,50 @@
 package de.rettichlp.shantybot.common.services;
 
-import de.rettichlp.shantybot.common.configuration.DiscordBotProperties;
 import lombok.extern.log4j.Log4j2;
-import net.dv8tion.jda.api.entities.Guild;
 import net.dv8tion.jda.api.entities.Role;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 
-import java.util.Timer;
-import java.util.TimerTask;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
 
+import static de.rettichlp.shantybot.ShantyBot.discordBotProperties;
 import static java.lang.System.currentTimeMillis;
-import static java.util.concurrent.TimeUnit.HOURS;
+import static java.util.Objects.nonNull;
+import static java.util.Optional.ofNullable;
+import static java.util.concurrent.CompletableFuture.allOf;
 
+@EnableScheduling
 @Log4j2
 @Component
 public class RoleSyncService {
 
-    public RoleSyncService(DiscordBotProperties discordBotProperties) {
-        long sixHoursInMillis = HOURS.toMillis(6);
+    @Scheduled(cron = "0 * * * * ?", zone = "Europe/Berlin") // every day at 0:00, 6:00, 12:00, 18:00 (UTC+1)
+    public void deleteOldLogEntries() {
+        long startTime = currentTimeMillis();
+        log.info("Discord role synchronising: started");
 
-        new Timer().scheduleAtFixedRate(new TimerTask() {
-            @Override
-            public void run() {
-                long startTime = currentTimeMillis();
-                log.info("Discord role synchronising: started");
+        ofNullable(discordBotProperties.getGuild()).ifPresentOrElse(guild -> {
+            Role memberRole = discordBotProperties.getMemberRole();
+            List<CompletableFuture<Void>> futures = guild.loadMembers().get().stream()
+                    .filter(member -> nonNull(memberRole))
+                    .filter(member -> !member.getUser().isBot())
+                    .filter(member -> !member.getRoles().contains(memberRole))
+                    .map(member -> {
+                        CompletableFuture<Void> future = new CompletableFuture<>();
+                        guild.addRoleToMember(member, memberRole).queue(success -> {
+                            log.info("Discord role synchronising: Add role {} to member {}", memberRole.getName(), member.getEffectiveName());
+                            future.complete(null);
+                            }, future::completeExceptionally);
+                        return future;
+                    })
+                    .toList();
 
-                Guild guild = discordBotProperties.getGuild();
-                guild.loadMembers().get().forEach(member -> {
-                    Role memberRole = discordBotProperties.getMemberRole();
-                    if (!member.getRoles().contains(memberRole)) {
-                        guild.addRoleToMember(member, memberRole).queue();
-                        log.info("Discord role synchronising: Add role {} to member {}", memberRole, member.getEffectiveName());
-                    }
-                });
-
-                log.info("Discord role synchronising: finished in {}ms", currentTimeMillis() - startTime);
-            }
-        }, (sixHoursInMillis - currentTimeMillis() % sixHoursInMillis) + HOURS.toMillis(2), sixHoursInMillis); // 4 10 16 22
-        log.info("Discord role synchronising: scheduled");
+            allOf(futures.toArray(new CompletableFuture[0])).thenRun(() -> log.info("Discord role synchronising: finished in {}ms", currentTimeMillis() - startTime)).exceptionally(error -> {
+                log.error("Discord role synchronising: failed - {}", error.getMessage());
+                return null;
+            });
+        }, () -> log.warn("Discord role synchronising: Skipped! Guild is null"));
     }
 }

--- a/src/main/java/de/rettichlp/shantybot/common/services/RoleSyncService.java
+++ b/src/main/java/de/rettichlp/shantybot/common/services/RoleSyncService.java
@@ -20,7 +20,7 @@ import static java.util.concurrent.CompletableFuture.allOf;
 @Component
 public class RoleSyncService {
 
-    @Scheduled(cron = "0 * * * * ?", zone = "Europe/Berlin") // every day at 0:00, 6:00, 12:00, 18:00 (UTC+1)
+    @Scheduled(cron = "0 0 */6 * * *", zone = "Europe/Berlin") // every day at 0:00, 6:00, 12:00, 18:00 (UTC+1)
     public void deleteOldLogEntries() {
         long startTime = currentTimeMillis();
         log.info("Discord role synchronising: started");

